v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/schema/SchemaStringOverridesParser.java

```diff
@@ -0,0 +1,173 @@
+   * @return A pair of spannerTableName and spannerColumnName
+   */
+  @Override
+  public String getColumnOverride(String sourceTableName, String sourceColumnName) {
+    return columnNameOverrides
+        .getOrDefault(
+            String.format("%s.%s", sourceTableName, sourceColumnName),
+            new ImmutablePair<>(sourceTableName, sourceColumnName))
+        .getRight();
+  }
+
+  private void parseTableMapping(String tableInput) {
+    String pairsString = tableInput.substring(1, tableInput.length() - 1);
+    int openBraceCount = 0;
+    int startIndex = 0;
+    for (int i = 0; i < pairsString.length(); i++) {
+      char c = pairsString.charAt(i);
+      if (c == '{') {
+        openBraceCount++;
+        if (openBraceCount == 1) {
+          startIndex = i;
+        }
+      } else if (c == '}') {
+        openBraceCount--;
+        if (openBraceCount == 0) {
+          String pair = pairsString.substring(startIndex + 1, i);
+          String[] tables = pair.split(",");
+          if (tables.length == 2) {
+            String sourceTable = tables[0].trim();
+            String destinationTable = tables[1].trim();
+            tableNameOverrides.put(sourceTable, destinationTable);
+          } else {
+            throw new IllegalArgumentException(
+                String.format(
+                    "Malformed pair encountered: %s, please fix the tableOverrides and rerun the job.",
+                    pair));
+          }
+        }
+      }
+    }
+  }
+
+  private void parseColumnMapping(String columnInput) {
+    String pairsString = columnInput.substring(1, columnInput.length() - 1);
+    int openBraceCount = 0;
+    int startIndex = 0;
+    for (int i = 0; i < pairsString.length(); i++) {
+      char c = pairsString.charAt(i);
+      if (c == '{') {
+        openBraceCount++;
+        if (openBraceCount == 1) {
+          startIndex = i;
+        }
+      } else if (c == '}') {
+        openBraceCount--;
+        if (openBraceCount == 0) {
+          String pair = pairsString.substring(startIndex + 1, i);
+          String[] tableColumnPairs = pair.split(",");
+          if (tableColumnPairs.length == 2) {
+            String[] sourceTableColumn = tableColumnPairs[0].trim().split("\\.");
+            String[] destTableColumn = tableColumnPairs[1].trim().split("\\.");
+            if (sourceTableColumn.length == 2 && destTableColumn.length == 2) {
+              columnNameOverrides.put(
+                  String.format("%s.%s", sourceTableColumn[0], sourceTableColumn[1]),
+                  new ImmutablePair<>(destTableColumn[0], destTableColumn[1]));
+            } else {
+              throw new IllegalArgumentException(
+                  String.format(
+                      "Malformed pair encountered: %s, please fix the columnOverrides and rerun the job.",
+                      pair));
+            }
+          } else {
+            throw new IllegalArgumentException(
```

Comment:
we should test for invalid scenarios as well, more important considering these are user inputs.

Reply:
User input is gated via regex. If any malformed string which slips through the regex will be caught in this exception which has been unit tested.